code/modules/projectiles/gun.dm

```diff
@@ -179,12 +179,9 @@
 			to_chat(user, SPAN_DANGER("The gun's safety is on!"))
 			handle_click_empty(user)
 			return FALSE
-	if(twohanded)
-		if(!wielded)
-			if (world.time >= recentwield + 1 SECONDS)
-				to_chat(user, SPAN_DANGER("The gun is too heavy to shoot in one hand!"))
-				recentwield = world.time
-			return FALSE
+	
+	if(twohanded_check(M))
```

Comment:
Have it so that if twohanded_check returns false it fails to fire. It's more intuitive.

Reply:
Ok, done.